src/intervals/functions.jl

```diff
@@ -142,25 +142,59 @@ function ^(a::Interval{Rational{T}}, x::AbstractFloat) where T<:Integer
 end
 
 # Rational power
-function ^(a::Interval{BigFloat}, r::Rational{S}) where S<:Integer
-    T = BigFloat
+function ^(a::Interval{T}, x::Rational) where T
     domain = Interval{T}(0, Inf)
 
+    p = x.num
+    q = x.den
+
+    isempty(a) && return emptyinterval(a)
+    x == 0 && return one(a)
+
     if a == zero(a)
-        a = a ∩ domain
-        r > zero(r) && return zero(a)
+        x > zero(x) && return zero(a)
         return emptyinterval(a)
     end
 
-    isinteger(r) && return atomic(Interval{T}, a^round(S,r))
-    r == one(S)//2 && return sqrt(a)
+    x == (1//2) && return sqrt(a)
+
+    if x >= 0
+        if a.lo ≥ 0
+            isinteger(x) && return a ^ Int64(x)
+            a = @biginterval(a)
+            ui = convert(Culong, q)
+            low = BigFloat()
+            high = BigFloat()
+            ccall((:mpfr_rootn_ui , :libmpfr) , Int32 , (Ref{BigFloat}, Ref{BigFloat}, Culong, MPFRRoundingMode) , low , a.lo , ui, MPFRRoundDown)
```

Comment:
This line is the `nthroot` function that we have been discussing. Please refactor it out into an `nthroot` function.